Content.Server/TapeRecorder/CassetteTapeSystem.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            base.Initialize();
+            SubscribeLocalEvent<CassetteTapeComponent, DamageChangedEvent>(OnDamagedChanged);
+            SubscribeLocalEvent<CassetteTapeComponent, InteractUsingEvent>(OnInteractUsing);
+            SubscribeLocalEvent<CassetteTapeComponent, SpoolingCompleteEvent>(OnSpoolingComplete);
+            SubscribeLocalEvent<CassetteTapeComponent, SpoolingCancelEvent>(OnSpoolingCancel);
+        }
+
+
+        /// <summary>
+        /// Handles "fixing" the cassette tape by screwing it back in with a screwdriver (ADD SUPPORT FOR PENCILS WHEN THOSE ARE ADDED)
+        /// </summary>
+        private void OnInteractUsing(EntityUid uid, CassetteTapeComponent component, InteractUsingEvent args)
+        {
+            if (!component.Unspooled || component.CancelToken != null)
+                return;
+            if (!_tagSystem.HasAnyTag(args.Used, "Screwdriver", "Write"))
+                return;
+
+            component.CancelToken = new CancellationTokenSource();
+
+            var doAfterEventArgs = new DoAfterEventArgs(args.User, 7, component.CancelToken.Token, component.Owner)
+            {
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = true,
+                TargetFinishedEvent = new SpoolingCompleteEvent
+                {
+                    User = args.User,
+                },
+                TargetCancelledEvent = new SpoolingCancelEvent()
+            };
+
+            _popupSystem.PopupEntity(Loc.GetString("cassette-repair-start", ("item", component.Owner)), args.User, Filter.Entities(args.User));
+            _doAfterSystem.DoAfter(doAfterEventArgs);
+        }
+
+        private void OnSpoolingComplete(EntityUid uid, CassetteTapeComponent component, SpoolingCompleteEvent args)
+        {
+            _popupSystem.PopupEntity(Loc.GetString("cassette-repair-finish", ("item", component.Owner)), args.User, Filter.Entities(args.User));
+
+            _tagSystem.AddTag(component.Owner, "CassetteTape");
+            if (!TryComp<SpriteComponent>(component.Owner, out var spriteComponent))
+                return;
+            spriteComponent.LayerSetVisible(1, false);
+            component.Unspooled = false;
+            component.TimeStamp = 0; //may as well rewind it while fixing it
+            component.CancelToken = null;
+        }
+
+        private void OnSpoolingCancel(EntityUid uid, CassetteTapeComponent component, SpoolingCancelEvent args)
+        {
+            component.CancelToken = null;
+        }
+
+        /// <summary>
+        /// Let the tape spaghetti fly out of the tape when it's damaged
+        /// </summary>
+        private void OnDamagedChanged(EntityUid uid, CassetteTapeComponent component, DamageChangedEvent args)
+        {
+            if (args.DamageDelta == null || args.DamageDelta.Total < 5)
+                return;
```

Comment:
Shouldn't this check if it's already unspooled.

Reply:
I wanted to have it so every time the tape gets hit, more data is erased. So i'm not checking if it's already unspooled